lib/commands/envs.ts

```diff
@@ -128,50 +89,23 @@ export default class EnvsCmd extends Command {
 		(for example, in case the current user has been removed from the application
 		by its owner).
 `;
-	public static examples = isV12()
-		? [
-				'$ balena envs --application MyApp',
-				'$ balena envs --application MyApp --json',
-				'$ balena envs --application MyApp --service MyService',
-				'$ balena envs --application MyApp --service MyService',
-				'$ balena envs --application MyApp --config',
-				'$ balena envs --device 7cf02a6',
-				'$ balena envs --device 7cf02a6 --json',
-				'$ balena envs --device 7cf02a6 --config --json',
-				'$ balena envs --device 7cf02a6 --service MyService',
-		  ]
-		: [
-				'$ balena envs --application MyApp',
-				'$ balena envs --application MyApp --all --json',
-				'$ balena envs --application MyApp --service MyService',
-				'$ balena envs --application MyApp --all --service MyService',
-				'$ balena envs --application MyApp --config',
-				'$ balena envs --device 7cf02a6',
-				'$ balena envs --device 7cf02a6 --all --json',
-				'$ balena envs --device 7cf02a6 --config --all --json',
-				'$ balena envs --device 7cf02a6 --all --service MyService',
-		  ];
+	public static examples = [
+		'$ balena envs --application MyApp',
+		'$ balena envs --application MyApp --json',
+		'$ balena envs --application MyApp --service MyService',
+		'$ balena envs --application MyApp --service MyService',
+		'$ balena envs --application MyApp --config',
+		'$ balena envs --device 7cf02a6',
+		'$ balena envs --device 7cf02a6 --json',
+		'$ balena envs --device 7cf02a6 --config --json',
+		'$ balena envs --device 7cf02a6 --service MyService',
+	];
 
 	public static usage = (
 		'envs ' + new CommandHelp({ args: EnvsCmd.args }).defaultUsage()
 	).trim();
 
 	public static flags: flags.Input<FlagsDef> = {
-		...(isV12()
-			? {
-					all: flags.boolean({
-						description: stripIndent`
-				No-op since balena CLI v12.0.0.`,
-						hidden: true,
```

Comment:
The flag is hidden, but should not yet be deleted from source code? If a CLI v12 user currently provides `--all` on the command line (because they had been doing it in their scripts since v11), it used to work before this PR, and should still work after this PR... (Otherwise, it's a breaking change.) If my reasoning is correct, you could use the newly added `isV13()` to conditionally add the hidden flag.

Btw, something else I remember we want for `isV13()` is to remove the `--gitignore` flag for the push / build / deploy commands... (Not saying you should do it in this PR, unless you want to...) :-)

Reply:
Yes, you are right.  I'll change to v13 switch.

Reply:
Ah no, I remember now.   Prior to this PR we had this line:
```
		options.all = options.all || isV12();
```
Since v12, `all` has been forced on, and the CLI flag value is ignored.  So, this PR shouldn't change behaviour.

Reply:
Except... Now adding the `-all` flag would cause an `unexpected argument` error.  Ok, so yeah I think we need to leave the flag in until v13, but it's safe to remove all the logic.